plugins/vpc-bridge/config/k8s_connector.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector
+
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-plugins/network/vpc"
+
+	winio "github.com/Microsoft/go-winio"
+	"github.com/Microsoft/go-winio/pkg/guid"
+	log "github.com/cihub/seelog"
+)
+
+// Represents output of k8s connector received from named pipe
+type k8sExecutionResult struct {
+	stdout string // Represents message with required data.
+	stderr error  // Represents error while getting output.
+}
+
+const (
+	// Represents timeout for reading over results channel.
+	// Represents timeout for reading from connection object.
+	k8sExecutionTimeout = 3 * time.Minute
+)
+
+func init() {
+	retrievePodConfigHandler = retrievePodConfig
```

Comment:
Why is this in init? Just assign the func

Reply:
done

Reply:
This failed while testing CNI plugin as `retrievePodConfigHandler func(netConfig *NetConfig) error` is defined in same package and it is required when build flags are not given `//go:build enablek8sconnector && windows`.

Reply:
Dumb. Can you please add a comment that says why its being assigned this way

Reply:
done